draft-bar-cfrg-spake2plus.md

```diff
@@ -233,43 +275,40 @@ specific protocol is given in {{?I-D.ietf-mmusic-sdp-uks}}).
-verification of B's confirmation message, A MUST send B a confirmation message
-using a MAC computed equivalently except with the use of pB and KcA. B MUST NOT
-send application data to A until it has received and verified the confirmation
-message. Key confirmation verification requires recomputation of the MAC and
-checking for equality against that which was received.
+so both parties can confirm that they agree upon these shared secrets. After
+receipt and verification of B's confirmation message, A MUST send B an equivalent
+confirmation message. B MUST NOT send application data to A until it has received
+and verified the confirmation message. Key confirmation verification requires
+recomputation of cA or cB and checking for equality against that which was received.
 
 # Key Schedule and Key Confirmation {#keys}
 
 The protocol transcript TT, as defined in {{spake2plus}},
-is unique and secret to A and B. Both parties use TT to
-derive shared symmetric secrets Ke and Ka as Ke || Ka = Hash(TT). The length of each
-key is equal to half of the digest output, e.g., |Ke| = |Ka| = 128 bits for SHA-256.
-If the required key size is less than half the digest output, e.g. when using SHA-512
-to derive two 128-bit keys, the digest output MAY be truncated.
+is unique and secret to A and B. Both parties use TT to derive shared symmetric
+secrets Ke and Ka. The length of each key is equal to half of the digest output,
+e.g., |Ke| = |Ka| = 128 bits for SHA-256. If the required key size is less than
+half the digest output, e.g. when using SHA-512 to derive two 128-bit keys, the
+digest output MAY be truncated.
 
-Both endpoints use Ka to derive subsequent MAC keys for key confirmation messages.
-Specifically, let KcA and KcB be the MAC keys used by A and B, respectively.
-A and B compute them as KcA || KcB = KDF(nil, Ka, "ConfirmationKeys")
+~~~
+Ka || Ke = Hash(TT)
+KcA || KcB = KDF(nil, Ka, "ConfirmationKeys")
+~~~
 
-The length of each of KcA and KcB is equal to half of the KDF
-output, e.g., |KcA| = |KcB| = 128 bits for HKDF-SHA256. If half of the KDF
-output size exceeds the required key size for the chosen MAC, e.g. when using
-HKDF-SHA512 as the KDF and CMAC-AES-128 as the MAC, the KDF output MAY be truncated.
+A and B output Ke as the shared secret from the protocol. Ka and its derived
+KcA and KcB are not used for anything except key confirmation and must be
+discarded after the protocol execution.
 
-The resulting key schedule for this protocol, given transcript TT, is as follows.
+Both endpoints MUST either exchange cA=KcA and cB=KcB directly, or employ a
+secure PRF, acting as a MAC that produces pseudorandom tags, for key confirmation.
+In the latter case, KcA and KcB are the symmetric keys used to compute tags cA
+and cB over the received key share of the respective peer.
 
 ~~~
-TT -> Hash(TT) = Ka || Ke
-Ka -> KDF(nil, Ka, "ConfirmationKeys") = KcA || KcB
+cA = MAC(KcA, pB)
```

Comment:
Is the input to this MAC actually relevant, if the alternative is to just send cA or cB directly? If not, I would simplify and make this MAC(KcA, <fixed string>). Rationale: encoding public keys is not something we really discuss, so it's just simpler to avoid it altogether, I think.

Reply:
It's not relevant. I've abstracted this part a bit.